src/query/query.pest

```diff
@@ -1,24 +1,34 @@
 // See the pest editor https://pest.rs/#editor for an easy way to see how
 // inputs are parsed by this grammar
 
-query         =  { SOI ~ expr ~ EOI }
-expr = _{ singleVariableFunc | noVariableFunc | infixExpression }
+query = { SOI ~ body ~ EOI }
+body  = { prefix? ~ expr ~ (booleanOperator ~ prefix? ~ expr)* }
+
+prefix   = _{ negation }
+negation = { "!" }
+
+booleanOperator = _{ and | or }
+and             = { "&&" }
+or              = { "||" }
```

Comment:
I wondered about that too, particularly if we're letting people enter search terms into Packit - though I guess we'd probably prefer to make a nice UI for that rather than making people type these things in raw, thought that could be the Advanced option...

Reply:
I'd rather not at this point, as then we will also have to write a parser to cope with the R version (we may have to at some point but there are some other interface issues here, such as allowing R users to use `quote()` which means that and/or will not be supported in some contexts)